spark/src/main/java/org/datacleaner/spark/AnalyzerResultReduceFunction.java

```diff
@@ -0,0 +1,102 @@
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.spark;
+
+import java.util.Arrays;
+
+import org.apache.spark.api.java.function.Function2;
+import org.datacleaner.api.AnalyzerResult;
+import org.datacleaner.api.AnalyzerResultReducer;
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.InjectionManager;
+import org.datacleaner.descriptors.ComponentDescriptor;
+import org.datacleaner.descriptors.Descriptors;
+import org.datacleaner.descriptors.ResultDescriptor;
+import org.datacleaner.job.ComponentJob;
+import org.datacleaner.lifecycle.LifeCycleHelper;
+
+public final class AnalyzerResultReduceFunction implements
+        Function2<NamedAnalyzerResult, NamedAnalyzerResult, NamedAnalyzerResult> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SparkJobContext _sparkJobContext;
+
+    public AnalyzerResultReduceFunction(SparkJobContext sparkJobContext) {
+        _sparkJobContext = sparkJobContext;
+    }
+
+    @Override
+    public NamedAnalyzerResult call(NamedAnalyzerResult namedAnalyzerResult1, NamedAnalyzerResult namedAnalyzerResult2)
+            throws Exception {
+
+        assert namedAnalyzerResult1.getName().equals(namedAnalyzerResult2.getName());
+
+        String key = namedAnalyzerResult1.getName();
+
+        ComponentJob componentJob = _sparkJobContext.getComponentByKey(key);
+
+        final AnalyzerResult analyzerResult1 = namedAnalyzerResult1.getAnalyzerResult();
+        final AnalyzerResult analyzerResult2 = namedAnalyzerResult2.getAnalyzerResult();
+
+        ResultDescriptor rd = getResultDescriptor(componentJob, analyzerResult1);
+        final Class<? extends AnalyzerResultReducer<?>> resultReducerClass = rd.getResultReducerClass();
+
+        if (resultReducerClass == null) {
+            throw new IllegalStateException("The analyzer (" + analyzerResult1 + ") is not distributable!");
+        }
+
+        AnalyzerResultReducer<AnalyzerResult> reducer = initializeReducer(resultReducerClass);
+        
+        AnalyzerResult reducedAnalyzerResult = reducer.reduce(Arrays.asList(analyzerResult1, analyzerResult2));
+
+        NamedAnalyzerResult reducedTuple = new NamedAnalyzerResult(key, reducedAnalyzerResult);
+        return reducedTuple;
+    }
+
+    private AnalyzerResultReducer<AnalyzerResult> initializeReducer(
+            final Class<? extends AnalyzerResultReducer<?>> resultReducerClass) {
+
+        final DataCleanerConfiguration configuration = _sparkJobContext.getConfiguration();
+        final InjectionManager injectionManager = configuration.getEnvironment().getInjectionManagerFactory()
+                .getInjectionManager(configuration, _sparkJobContext.getAnalysisJob());
+        final LifeCycleHelper lifeCycleHelper = new LifeCycleHelper(injectionManager, true);
```

Comment:
That was a copy and paste, will fix it. However, if there are non-distributable analyzers in the job, it will throw an exception before it even gets to this line.